File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/Spectacles.java
package org.rexcellentgames.burningknight.entity.item.accessory.equippable;

import org.rexcellentgames.burningknight.entity.creature.player.Player;
import org.rexcellentgames.burningknight.assets.Locale;

public class Spectacles extends Equippable {
	{
		name = Locale.get("spectacles");
		description = Locale.get("spectacles_desc");
		sprite = "item-spectacles";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);

		if (this.owner instanceof Player) {
			((Player) this.owner).seeSecrets = true;
		}
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);

		if (this.owner instanceof Player) {
			((Player) this.owner).seeSecrets = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/Spectacles.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class Spectacles : Equippable() {
	init {
		name = Locale.get("spectacles")
		description = Locale.get("spectacles_desc")
		sprite = "item-spectacles"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)

		this.owner.seeSecrets = true
		this.owner.checkSecrets()
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)

		this.owner.seeSecrets = false
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}

	override fun canBeDegraded(): Boolean {
		return false
	}
}